test/games/strategy/engine/data/UnitCollectionTest.java

```diff
@@ -8,43 +8,63 @@
 import java.util.Collection;
 import java.util.Iterator;
 
+import org.junit.Before;
 import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
 
 import games.strategy.triplea.delegate.Matches;
 import games.strategy.util.IntegerMap;
 
 public class UnitCollectionTest {
 
-  final GameData gameData = new MyGameData();
-  final UnitType unitTypeOne = new UnitType("Unit Type 1", gameData);
-  final UnitType unitTypeTwo = new UnitType("Unit Type 2", gameData);
-  final MyPlayerID defaultPlayerID = new MyPlayerID();
-  final PlayerID otherPlayerID = PlayerID.NULL_PLAYERID;
-  final UnitCollection uc = new UnitCollection(defaultPlayerID, gameData);
+  @Mock
+  private final GameData gameData = Mockito.mock(GameData.class);
 
-  final int unitCountDefaultPlayerUnitTypeOne = getUnitsOfDefaultPlayerOfUnitTypeOne().size();
-  final int unitCountDefaultPlayerUnitTypeTwo = getUnitsOfDefaultPlayerOfUnitTypeTwo().size();
-  final int unitCountDefaultPlayer = unitCountDefaultPlayerUnitTypeOne + unitCountDefaultPlayerUnitTypeTwo;
-  final Unit unitDefaultPlayer1 = new Unit(unitTypeOne, defaultPlayerID, gameData);
-  final Unit unitDefaultPlayer2 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
-  final Unit unitDefaultPlayer3 = new Unit(unitTypeTwo, defaultPlayerID, gameData);
+  @Before
+  public void setup() {
+    final UnitTypeList unitTypeList = new UnitTypeList(gameData);
+    unitTypeList.addUnitType(unitTypeOne);
+    unitTypeList.addUnitType(unitTypeTwo);
+    Mockito.when(gameData.getUnitTypeList()).thenReturn(unitTypeList);
+  }
+
+  final private UnitType unitTypeOne = new UnitType("Unit Type 1", gameData);
+  final private UnitType unitTypeTwo = new UnitType("Unit Type 2", gameData);
+  @Mock
+  final private MyPlayerID defaultPlayerID = new MyPlayerID();

```

Comment:
Same again, @Mock will handle instantiating the object for you, no need to do the `new MyPlayerID()`